app/src/main/java/com/lagradost/cloudstream3/MainActivity.kt

```diff
@@ -1152,6 +1156,16 @@ class MainActivity : AppCompatActivity(), ColorPickerDialogListener {
 
         changeStatusBarState(isEmulatorSettings())
 
+        /** Biometric stuff for users without accounts **/
+        val authEnabled = settingsManager.getBoolean(getString(R.string.biometric_enabled_key), false)
+        val noAccounts = settingsManager.getBoolean(getString(R.string.skip_startup_account_select_key), false) || accounts.count() <= 1
+
+        if (isTruePhone() && authEnabled && noAccounts ) {
+            BiometricAuthenticator.initializeBiometrics(this@MainActivity)
+            BiometricAuthenticator.checkBiometricAvailability()
+            BiometricAuthenticator.biometricPrompt.authenticate(promptInfo)
+        }
+
```

Comment:
Nice patch, but this can still be kept to be done in AccountSelectActivity, before we navigate to the main activity if we are skipping account selection. This would just prevent duplicate code. IE see how PIN is asked even if we are skipping account selection and a PIN exists on the last used account. A little difference but some type of premise for the most part if I am understanding this patch correctly.

Reply:
yes, it xan prompted on account selects navigation to main and works but when auth fails or dismissed there, it doesn't finishes the app, totally against authentication, as it requires oncreate of main activity to finish() it.